### ISSUE TITLE
Return types in sub-classed class docstrings need to be fully qualified for intersphinx to work

### ISSUE THREAD

**Issue by Andy-Grigg** [pyansys/openapi-common#124]
Discovered while working on https://github.com/pyansys/grantami-bomanalytics/issues/96

It looks like for the docs of the bomanalytics subclass to correctly interpret the docstrings of the superclass, they need to be fully qualified.

For example, the `ApiClientFactory.connect` method has a return type of :class:`ApiClient`. However, the docs for bomanalytics for `Connection(ApiClientFactory)` do not correctly pick up that the references to classes are referring to things in the `ansys.openapi.common` namespace. Sphinx would have to inspect the class hierarchy to figure this out, which I'm assuming it doesn't do.

If the docstring is changed to :class:`~ansys.openapi.common.ApiClient` then everything works as expected.

Assuming this is a change that we want to make (I don't think we have much of a choice though if we want intersphinx mappings to work), this will need to be added to the following docstrings:

- ApiClientFactory.with_autologon return type
- ApiClientFactory.with_credentials return type
- ApiClientFactory.with_oidc idp_session_configuration type (SessionConfiguration)
- ApiClientFactory.with_oidc return type (OIDCSessionBuilder)
- ApiClientFactory.connect return type (ApiClient)

Basically anything that bomanalytics refers to.
